lib/query_parser.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+# Common space for Insights API stuff
+module Insights
+  module API
+    module Common
+      # Query parser that can handle insights-flavored tags
+      class QueryParser < ::SimpleDelegator
```

Comment:
Tests please? :slightly_smiling_face:

Reply:
It's actually tested as a systems query, but sure I can write some unit tests :wink: